### PR TITLE
Allow auto generated password usage

### PR DESCRIPTION
**Pull request by HNicolas**
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Special notes for your reviewer:
It should now be possible to deploy a replicaset specifying a mongodbUsername and mongodbDatabase but using without specifying a password for the user (the chart should generate the secret).

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/mongodb/templates/secrets.yaml [hunk 1]
```diff
@@ -10,13 +10,11 @@ metadata:
     heritage: "{{ .Release.Service }}"
 type: Opaque
 data:
-  {{- if .Values.usePassword }}
```
Comment (juan131) [inline stable/mongodb/templates/secrets.yaml:13]: Why do you remove this condition?

File: stable/mongodb/templates/statefulset-primary-rs.yaml [hunk 2]
```diff
@@ -111,7 +111,7 @@ spec:
             value: {{ .Values.mongodbDatabase | quote }}
           {{- end }}
             {{- if .Values.usePassword }}
-            {{- if and .Values.mongodbUsername (or .Values.mongodbPassword .Values.existingSecret) }}
+            {{- if and .Values.mongodbUsername .Values.mongodbDatabase }}
```
Comment (juan131) [inline stable/mongodb/templates/statefulset-primary-rs.yaml:114]: This will affect supporting existing secrets. Please revert this change.

**Review by juan131 — CHANGES_REQUESTED** [review]
Hi @HNicolas 

Thanks for the contribution!! 

Could you please provide more context about why you need the password to be randomly created? I understand that root password can be created this way... but why is it necessary for a password which is exclusively used for a custom user/database created during the initialization?

File: stable/mongodb/templates/secrets.yaml (hunk 1 above)
Comment (HNicolas) [reply stable/mongodb/templates/secrets.yaml:13]: Hi @juan131,
This condition is useless as this is already checked on the first line of the file.

File: stable/mongodb/templates/statefulset-primary-rs.yaml (hunk 2 above)
Comment (HNicolas) [reply stable/mongodb/templates/statefulset-primary-rs.yaml:114]: I used the same logic than in the already existing secret file : if the user provide mongodbUsername and a mongodbDatabase the password must be provided to.
You can see on line 118 of the file that it will not affect existing secrets support because if an existing secret is provided the password is read from it.

I just want the chart to generate a secret for me if not provided. The logic was already implemented in the secrets.yaml file but it didn't work because of the line 114 of statefulset-primary-rs.yaml.

File: stable/mongodb/templates/secrets.yaml (hunk 1 above)
Comment (juan131) [reply stable/mongodb/templates/secrets.yaml:13]: You're right!

File: stable/mongodb/templates/statefulset-primary-rs.yaml (hunk 2 above)
Comment (juan131) [reply stable/mongodb/templates/statefulset-primary-rs.yaml:114]: You're right, it doesn't affect existing secrets. 

Regarding generating random passwords when not provided, I don't find it useful since it's a password for a custom user/database (so I'm supposed to know what password I want to use). However, I think it's makes no harm to support that feature.

**Review by juan131 — APPROVED** [review]
Approving your changes!
